### PR TITLE
feat(routes): signup timestamps

### PR DESCRIPTION
**Pull request by simonyangme**
closes hoodiehq/camp#61

@gr2m please review

### CODE REVIEW

File: routes/account.js [hunk 1]
```diff
@@ -36,9 +36,17 @@ function accountRoutes (server, options, next) {
       var password = request.payload.data.attributes.password
       var id = request.payload.data.id
       var query = request.query
+
+      var currentTime = new Date().toISOString()
+      var createdAt = request.payload.data.attributes.createdAt
+      if (!createdAt) { // If createdAt is not assigned, use current time
+        createdAt = currentTime
+      }
```
Comment (gr2m) [inline routes/account.js]: can you move `var createdAt = request.payload.data.attributes.createdAt` to the top of the function (bellow `var password = ...` and then replace `var createdAt = request.payload.data.attributes.createdAt` with `var createdAt = createdAt || currentTime`

Comment (simonyangme) [inline routes/account.js]: Done

File: routes/account.js [hunk 2]
```diff
@@ -34,11 +34,18 @@ function accountRoutes (server, options, next) {
...
       var query = request.query
+
+      var currentTime = new Date().toISOString()
+      var createdAt = createdAt || currentTime
+
       accounts.add({
         username: username,
         password: password,
+        createdAt: createdAt,
+        signedUpAt: currentTime,
```
Comment (gr2m) [inline routes/account.js]: put `createdAt || currentTime` directly in this line and remove 42

Comment (simonyangme) [inline routes/account.js]: Done

**Comment by gr2m** [comment]
I wonder where we should document that? This goes beyond the current [JSON API SPEC](http://docs.accountjsonapi.apiary.io). One way would be to extend the documentation in [routes/README.md](https://github.com/hoodiehq/hoodie-account-server/tree/master/routes/README.md). Another possibility would be to make the JSON API SPEC custom to Hoodie only and put it all in there. Originally I thought the JSON API SPEC might be nice to be implemented by others as well, but that’s not going to happen so I see no reason to rename it from account-json-api to hoodie-account-json-api and put all our additions in there for the route documentation

Any thoughts?

**Comment by simonyangme** [comment]
Does this need to be separate from the docs on apiary? Needing to go to two places to see the full picture is not the best. If the current account-json-api spec isn't going to be adopted by others, then I think it'd be best to just add it to the spec so that the single source of truth is reliable.

**Comment by gr2m** [comment]
yeah I agree, let’s just have one single source of truth. I’d suggest to put it into https://github.com/hoodiehq/account-json-api for now, but eventually we just probably merge that repository into `hoodie-account-server`

**Pull request merged**